src/chain_table.rs

```diff
@@ -21,121 +21,154 @@ use std::{
 };
 
 use cosmian_crypto_core::{
-    bytes_ser_de::{Deserializer, Serializer},
+    bytes_ser_de::{Deserializer, Serializable, Serializer},
     reexport::rand_core::CryptoRngCore,
     symmetric_crypto::{Dem, SymKey},
 };
 
 use crate::{
     error::CoreError as Error,
-    structs::{Block, Uid},
+    structs::{Block, BlockPrefix, BlockType, Uid},
     KeyingMaterial, CHAIN_TABLE_KEY_DERIVATION_INFO,
 };
 
 /// Value of the Chain Table. It is composed of a maximum of `TABLE_WIDTH`
 /// blocks of length `BLOCK_LENGTH`.
 #[must_use]
-#[derive(Clone, Debug, Default, PartialEq, Eq)]
-pub struct ChainTableValue<const BLOCK_LENGTH: usize>(Vec<Block<BLOCK_LENGTH>>);
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ChainTableValue<const TABLE_WIDTH: usize, const BLOCK_LENGTH: usize> {
+    length: usize,
```

Comment:
what does `length` represent here? the number of blocks?

Reply:
Yes, since I switched for a fixed-size representation of the `ChainTableValue`, I need a way to keep track of the number of blocks that are not padding. This is not part of the serialization so no overhead is added, but having a fixed-size object allows for some performance improvements.